src/gpio_monitor.cpp

```diff
@@ -23,7 +25,48 @@ void GpioEventHandler::setEventHandlerForGpioPresence(
 void GpioMonitor::initHandlerForGpio(
     const std::shared_ptr<boost::asio::io_context>& i_ioContext)
 {
-    // ToDo Add implementation.
-    (void)i_ioContext;
+    if (!m_sysCfgJsonObj.contains("frus"))
+    {
+        logging::logMessage("Frus section is missing in config JSON");
+        return;
+    }
+
+    try
+    {
+        for (const auto& l_Fru : m_sysCfgJsonObj["frus"].items())
```

Comment:
Following code suits better under JSON utility.
Please see if creating these APIs are feasible or makes more sense.
a) isPollingRequired() - generic API to check if a FRU qualifies for polling
b) getPollingParameters() - API to return parameters from JSON required to register a callback.

Reply:
Updated